pandas/tests/indexing/test_indexing.py

```diff
@@ -4035,11 +4035,11 @@ def f():
 
         self.assertRaises(ValueError, f)
 
-        # these are coerced to float unavoidably (as its a list-like to begin)
+        # these are coerced to object unavoidably (as its a list-like to begin)
         df = DataFrame(columns=['A', 'B'])
         df.loc[3] = [6, 7]
         assert_frame_equal(df, DataFrame(
-            [[6, 7]], index=[3], columns=['A', 'B'], dtype='float64'))
+            [[6, 7]], index=[3], columns=['A', 'B'], dtype='object'))
```

Comment:
Can you please elaborate some more?  I am not sure what you mean, and why `float64` makes sense.  Just trying to understand